### PR TITLE
Add dialog for selecting trait

### LINKED ISSUE #633: Replace trait dropdown in Collect with dialog

### PR DESCRIPTION
**Pull request by kamathprasad9**
## Description

Fixes #633 
Use a dialog to pick traits instead of spinner

## Type of change

_What type of changes does your code introduce? Put an `x` in boxes that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [x] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated relevant documentation

## Changelog entry

```release-note

```

### CODE REVIEW

**Comment by trife** [comment]
- [ ] Add neutral button "Modify" that opens the Traits activity
- [ ] Change OK button to Cancel
- [ ] Highlight currently selected trait in the list
- [ ] Left align the list items
- [ ] Wrap text (with tab indented second lines) for very long trait names

**Review by trife — APPROVED** [review]

**Comment by trife** [comment]
I made a few minor changes and this looks good to me. It can be merged once @chaneylc reviews.

File: app/src/main/java/com/fieldbook/tracker/views/TraitBoxView.kt [hunk 1]
```diff
@@ -344,9 +393,9 @@ class TraitBoxView : ConstraintLayout {
 
         val rangeBox = controller.getRangeBox()
         if (direction == "left") {
-            pos = traitType.selectedItemPosition - 1
+            pos = getSelectedItemPosition() - 1
             if (pos < 0) {
-                pos = traitType.count - 1
+                pos = visibleTraitsList!!.count() - 1
                 if (controller.isCyclingTraitsAdvances()) {
```
Comment (chaneylc) [inline app/src/main/java/com/fieldbook/tracker/views/TraitBoxView.kt:399]: Check for nullability here or try/catch this !! operator usage.

**Review by chaneylc — CHANGES_REQUESTED** [review]
Looks good, a few places you are using !! operator without try/catching or using null checks, please update that and I will accept.

**Review by chaneylc — APPROVED** [review]

**Pull request merged**